src/components/Form/Form.js

```diff
@@ -10,33 +10,49 @@ const Form = ({ createNewUser, loginUser, updateLocation, error }) => {
     const [address, setAddress] = useState('')
     const [city, setCity] = useState('')
     const [zip, setZip] = useState('')
+    const [inputError, setInputError] = useState('')
    
     const handleSubmit = e => {
-        if (location === '/newuser' && first && last && address && city ) {
+        if (location === '/newuser' && first && last && address && city && zipAuth(zip) ) {
             createNewUser(userName, first, last, address, city, zip)
             clearInputs()
-        } else if (location === '/login' && userName && !error) {
+        } else if (location === '/login' && userName) {
             loginUser(userName)
             clearInputs()
-        } else if (location === '/newlocation' && address && city) {
+        } else if (location === '/newlocation' && address && city && zipAuth(zip) ) {
             updateLocation(address, city, zip)
             clearInputs()
         } else {
             e.preventDefault()
+            setInputError('Please Complete The Form Below.')
         }
     }
 
+    const zipAuth = zip => {
+        console.log(zip)
+        const canada = /^[ABCEGHJ-NPRSTVXY]\d[ABCEGHJ-NPRSTV-Z][ -]?\d[ABCEGHJ-NPRSTV-Z]\d$/i
+        const us = /^[0-9]{5}(?:-[0-9]{4})?$/
+        console.log((canada.exec(zip) || us.exec(zip)? true : false))
+        return (canada.exec(zip) || us.exec(zip) ? true : false)
+        
```

Comment:
should delete these console.logs

Reply:
Will do. Also not sure why this test is failing on travis. It is passing locally.